### PR TITLE
Image urls

### PR DESCRIPTION
**Pull request by gestchild**
Prepares for Prismic switching on its imgix service.

Using the imgix service has page weight improvements:

Before:
story_before
After:
story_after

### CODE REVIEW

File: common/utils/convert-image-uri.js [hunk 1]
```diff
@@ -5,36 +5,32 @@ const imageMap = {
...
     iiifRoot: 'https://iiif.wellcomecollection.org/image/',
-    iiifOriginRoot: 'https://iiif-origin.wellcomecollection.org/image/',
   },
   iiif: {
     // sometimes we already have the iiif url, but we may want to convert it to use the origin
     root: 'https://iiif.wellcomecollection.org/image/',
     iiifRoot: 'https://iiif.wellcomecollection.org/image/',
-    iiifOriginRoot: 'https://iiif-origin.wellcomecollection.org/image/',
   },
 };
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]: There's a lot we could remove from here it looks like...

File: common/utils/convert-image-uri.js [hunk 2]
```diff
@@ -45,94 +41,148 @@ function determineSrc(url: string): string {
...
+  if (originalUriPath.includes('.png')) {
     return 'png';
   } else {
     return 'jpg';
   }
 }
 
-function convertPathToWordpressUri(originalUriPath, size) {
-  return originalUriPath + `?w=${size}`;
+type prismicUriOpts = {
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]: Exact typeing and PascalCase?

File: common/utils/convert-image-uri.js [hunk 3]
```diff
@@ -45,94 +41,148 @@ function determineSrc(url: string): string {
...
+    acc[key] = value;
+    return acc;
+  }, {});
+}
+
+function prismicTemplateParts(
+  originalUri: string,
+  requiredSize: number | 'full'
+) {
+  const uriParts = originalUri.split('?');
```
[the remark this answers is not in the document]
Comment (gestchild) [reply common/utils/convert-image-uri.js]: looks like we can yes - much nicer, thanks

File: common/utils/convert-image-uri.js [hunk 4]
```diff
@@ -45,94 +41,147 @@ function determineSrc(url: string): string {
...
+  };
 }
 
-function convertPathToIiifUri(originalUriPath, iiifRoot, size) {
-  const isFullSize = size === 'full';
-  const format = determineFinalFormat(originalUriPath);
-  return `${iiifRoot}${originalUriPath}/full/${size}${
-    isFullSize ? '' : ','
-  }/0/default.${format}`;
+type wordpressUriOpts = {|
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]: `s/w/W`

File: common/utils/convert-image-uri.js [hunk 5]
```diff
@@ -45,94 +41,147 @@ function determineSrc(url: string): string {
...
-export function convertIiifUriToInfoUri(originalUriPath: string) {
-  const match = originalUriPath.match(
-    /^https:\/\/iiif\.wellcomecollection\.org\/image\/(.+?\.[a-z]{3})/
-  );
-  if (match && match[0]) {
-    return `${match[0]}/info.json`;
-  } else {
-    return `${originalUriPath}/info.json`;
-  }
+type IiifUriOpts = {
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]: Exact type?

File: common/utils/convert-image-uri.js [hunk 6]
```diff
@@ -45,94 +41,147 @@ function determineSrc(url: string): string {
...
 }
 
-function convertPathToIiifUri(originalUriPath, iiifRoot, size) {
-  const isFullSize = size === 'full';
-  const format = determineFinalFormat(originalUriPath);
-  return `${iiifRoot}${originalUriPath}/full/${size}${
-    isFullSize ? '' : ','
-  }/0/default.${format}`;
+type WordpressUriOpts = {|
+  width?: number | 'full',
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]: It doesn't look as though width is optional though?

File: common/utils/convert-image-uri.js [hunk 7]
```diff
@@ -45,94 +41,147 @@ function determineSrc(url: string): string {
...
 }
 
-function convertPathToWordpressUri(originalUriPath, size) {
-  return originalUriPath + `?w=${size}`;
+type PrismicUriOpts = {|
+  auto?: string,
+  rect?: string,
+  w?: number | 'full',
+  h?: number,
+|};
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]: Same as below, are all fields optional?

File: common/utils/convert-image-uri.js [hunk 8]
```diff
@@ -45,94 +41,147 @@ function determineSrc(url: string): string {
...
+  const defaultOpts = {
+    region: 'full',
+    size: 'full',
+    rotation: 0,
+    quality: 'default',
+    format: 'jpg',
+  };
+  const template = urlTemplate.parse(templateString);
+  return (opts: IiifUriOpts) =>
+    template.expand(Object.assign({}, defaultOpts, opts));
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]: You could use spread.

File: common/utils/convert-image-uri.js (hunk 7 above)
Comment (gestchild) [reply common/utils/convert-image-uri.js]: yep

File: common/utils/convert-image-uri.js (hunk 6 above)
Comment (gestchild) [reply common/utils/convert-image-uri.js]: it would return an unsized image

**Pull request merged**